src/coord/src/coord/dataflow_builder.rs

```diff
@@ -101,9 +101,10 @@ impl<'a> DataflowBuilder<'a> {
                         dataflow.import_source(*id, source_description, persist_details);
                     }
                     CatalogItem::Source(source) => {
-                        if source.connector.requires_single_materialization() {
-                            let source_type =
-                                RematerializedSourceType::for_connector(&source.connector);
+                        if source.connector.requires_single_materialization()
+                            || source.persist_details.is_some()
```

Comment:
throw a "// A source may only be persisted once." (or something to that effect) comment here please

Reply:
Will do!

Reply:
I can't move it completely to the method on the connector, because that doesn't have the `persist_details`. I'll push a commit that adds a method `requires_single_materialization()` on `Source` that defers to the method on connector but also checks for persistence. How do you like that?